src/handlers/comment/handlers/review.ts

```diff
@@ -0,0 +1,196 @@
+    return `Error getting issue comments`;
+  }
+  streamlined.push({
+    login: issue.user.login,
+    body: issue.body,
+  });
+
+  comments.forEach(async (comment, i) => {
+    if (comment.user.type == UserType.User || commentsRaw[i].body.includes("<!--- { 'UbiquityAI': 'answer' } --->")) {
+      streamlined.push({
+        login: comment.user.login,
+        body: comment.body,
+      });
+    }
+  });
+
+  // returns the conversational context from all linked issues and prs
+  const links = await getAllLinkedIssuesAndPullsInBody(issue.number);
+
+  if (typeof links === "string") {
+    logger.info(`Error getting linked issues or prs: ${links}`);
+    return `Error getting linked issues or prs: ${links}`;
+  }
+  linkedIssueStreamlined = links.linkedIssues;
+  linkedPRStreamlined = links.linkedPrs;
+
+  // return a diff of the changes made in the PR
+  const comparePR = async () => {
+    const comparePR = await context.octokit.pulls.get({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      pull_number: issue.number,
+    });
+
+    const pr = comparePR.data;
+
+    const prDiff = await context.octokit.pulls.get({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      pull_number: pr.number,
+      mediaType: {
+        format: "diff",
+      },
+    });
+
+    const diffContent = prDiff.data;
+
+    return {
+      pr,
+      diff: diffContent,
+    };
+  };
+
+  const isPull = async () => {
+    if (isPr) {
+      const diff = await comparePR()
+        .then(({ diff }) => {
+          return diff;
+        })
+        .catch((error) => {
+          logger.info(`Error getting diff: ${error}`);
+          return `Error getting diff: ${error}`;
+        });
+
+      chatHistory.push({
+        role: "system",
+        content: gptContextTemplate,
+        name: "UbiquityAI",
+      } as CreateChatCompletionRequestMessage);
+
+      // We're allowing Chad to deduce what is the most relevant context
+      const gptDecidedContext = await decideContextGPT(chatHistory, streamlined, linkedPRStreamlined, linkedIssueStreamlined);
+
+      if (typeof gptDecidedContext === "string") {
+        return gptDecidedContext;
+      }
+
+      chatHistory = [];
+      chatHistory.push(
+        {
+          role: "system",
+          content: specCheckTemplate, // provide the spec check template
+          name: "UbiquityAI",
+        } as CreateChatCompletionRequestMessage,
+        {
+          role: "system",
+          content: "Context: \n" + JSON.stringify(gptDecidedContext.answer), // provide the context
+          name: "UbiquityAI",
+        } as CreateChatCompletionRequestMessage,
+        {
+          role: "user",
+          content: "PR Diff: \n" + JSON.stringify(diff), // provide the diff
+          name: "user",
+        } as CreateChatCompletionRequestMessage
+      );
+
+      const draftReport: GPTResponse | string = await askGPT("", chatHistory);
```

Comment:
Seems unusual to pass in an empty string

Reply:
The first arg is used only for logging purposes as it's the chatHistory which matters but I'll pass an identifier for logs.